source/GameInterface/Services/BesiegerCamps/Extensions/BesiegerCampExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿using GameInterface.Services.ObjectManager;
+using Serilog;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem.Siege;
+
+namespace GameInterface.Services.BesiegerCamps.Extensions
+{
+    public static class BesiegerCampExtensions
+    {
+        public static string TryGetId(object value, ILogger logger)
```

Comment:
Any "Try" prefix should return a bool

```suggestion
        public static bool TryGetId(object value, ILogger logger, out string id)
```

Reply:
Agreed, done.